### PR TITLE
fix: improve San'layn implementation for death knights

### PR DESCRIPTION
**Pull request by johnnylam88**
Improve the San'layn hero talents implementation from 11.0.5 for death knights, and catch up to some talent changes for death knights from 11.0.

### CODE REVIEW

File: TheWarWithin/DeathKnightBlood.lua [hunk 1]
```diff
@@ -1187,7 +1185,14 @@ spec:RegisterHook( "reset_precast", function ()
...
     end
 
-    if talent.vampiric_strike.enabled and IsActiveSpell( 433899 ) then applyBuff( "vampiric_strike" ) end
+    if talent.infliction_of_sorrow.enabled and buff.gift_of_the_sanlayn.up then
+        state:QueueAuraExpiration( "gift_of_the_sanlayn", TriggerInflictionOfSorrow, buff.gift_of_the_sanlayn.expires )
+    end
+
+    if IsActiveSpell( 433899 ) or IsActiveSpell( 433895 ) then
+        applyBuff( "vampiric_strike" )
+        class.abilities[ 433895 ] = class.abilities.heart_strike
```
Comment (Hekili) [inline TheWarWithin/DeathKnightBlood.lua]: Why here?  This normally goes in the `copy` line of the ability model (`RegisterAbilities`).

**Review by Hekili — CHANGES_REQUESTED** [review]
One small check.

File: TheWarWithin/DeathKnightBlood.lua (hunk 1 above)
Comment (johnnylam88) [reply TheWarWithin/DeathKnightBlood.lua]: You're right, that should be enough. I was mostly just copy-pasting from the unholy implementation for speed.

Comment (johnnylam88) [reply TheWarWithin/DeathKnightBlood.lua]: I'll just delete that assignment and verify everything still works.

**Comment by johnnylam88** [comment]
I've pushed one small commit to remove that unnecessary assignment. Testing on the target dummy shows that it's working correctly.

**Review by Hekili — APPROVED** [review]

**Pull request merged**